packages/typespec-azure-portal-core/src/lib.ts

```diff
@@ -0,0 +1,37 @@
+import { createTypeSpecLibrary } from "@typespec/compiler";
+
+export const $lib = createTypeSpecLibrary({
+  name: "@azure-tools/typespec-azure-portal-core",
+  diagnostics: {
+    fileNotFound: {
+      severity: "error",
+      messages: {
+        browseargQuery: `cannot find @browse file argQuery`,
+        aboutIcon: `cannot find @about file icon`,
+      },
+    },
+    invalidUsageDecorator: {
+      severity: "error",
+      messages: {
+        browse: `@browse decorator can be only applied to trackedResource and proxyResource`,
+        about: `@about decorator can be only applied to trackedResource and proxyResource`,
+        marketplaceOffer: `@marketPlaceOffer decorator can be only applied to trackedResource and proxyResource`,
+      },
+    },
+    invalidType: {
```

Comment:
Suggest `"invalid-offer-id"` or similar for the diagnostic name.

Reply:
fixed :) thank you!